addon/TelemetryRappor.jsm

```diff
@@ -318,44 +414,25 @@ var TelemetryRappor = {
    * @param {float} params.f - Value for probability f.
    * @param {float} params.p - Value for probability p.
    * @param {float} params.q - Value for probability q.
+   * @param {boolean} isSimulation - Boolean that indicates if the addon is run in a simulation.
+   * @param {int} simCohort - Cohort for the simulation.
    *
    * @return An object containing the cohort and the encoded value in hex.
    */
-  createReport(name, value, params) {
-    // Generate the RAPPOR secret. This secret never leaves the client.
-    let secret = null;
-    log.debug("HEYYY");
-    try {
-      secret = Services.prefs.getCharPref(PREF_RAPPOR_SECRET);
-      if (secret.length != 64) {
-        secret = null;
-      }
-    } catch (e) {
-      log.error("Error getting secret from prefs", e);
-    }
-    if (!secret) {
-      let randomArray = new Uint8Array(32);
-      crypto.getRandomValues(randomArray);
-      secret = bytesToHex(randomArray);
-      Services.prefs.setCharPref(PREF_RAPPOR_SECRET, secret);
-    }
-
-    // If we haven't self-selected a cohort yet for this measurement, then do so now,
-    // otherwise retrieve the cohort.
+  createReport(name, value, params, isSimulation = false, simCohort = null) {
+    let secret = getSecret(name);
     let cohort = null;
-    try {
-      cohort = Services.prefs.getIntPref(PREF_RAPPOR_PATH + name + ".cohort");
-    } catch (e) {
-      log.error("Error getting the cohort", e);
+    if (isSimulation) {
```

Comment:
Can't this be

`let cohort = isSimulation ? simCohort : getCohort(...);`

Reply:
You're right!